core/src/main/java/org/apache/gravitino/catalog/CatalogManager.java

```diff
@@ -738,45 +739,41 @@ public boolean dropCatalog(NameIdentifier ident, boolean force)
         metalakeIdent,
         LockType.WRITE,
         () -> {
-          checkMetalake(metalakeIdent, store);
```

Comment:
This change has the same effect as https://github.com/apache/gravitino/pull/9563, as `catalogInUse` will also access and check the existence of Metalake. see:

https://github.com/apache/gravitino/blob/bd41384a6c1b92a96a6aefb6c54c0ffb748a2bff/core/src/main/java/org/apache/gravitino/catalog/CatalogManager.java#L867-L871

The difference is that `checkMetalake` will throw an exception if a metalake is not in use, and `catalogInUse` will return false.

@mchades please help to verify whether this breaks your initial design

Reply:
You mean that it misses "metalake not in use exception" after my change? Let me check it. My initial feeling is that this part of code is quite messy and spread everywhere. It will be misused easily, we should figure out a better solution.